src/exporter/DaemonMetricCollector.cc

```diff
@@ -122,6 +122,9 @@ void DaemonMetricCollector::dump_asok_metrics() {
     for (auto &perf_group_item : counter_schema) {
       std::string perf_group = {perf_group_item.key().begin(),
                                 perf_group_item.key().end()};
+      if (!perf_group_item.value().is_array() && !counter_dump[perf_group].is_array()) {
```

Comment:
> If the perf_group_item isn't an array as_array() will throw invalid argument exception,
which will further lead to crashing of the exporter daemon.

Can you provide an example `counter schema` and/or `counter dump` output where either isn't an array?  Because https://github.com/ceph/ceph/pull/51947 was supposed to make both an array, in all cases.

Reply:
I think the crashing issue occurs when the cluster is under upgrade state. So exporter basically is using the old format for fetching counter dump/schema for ceph daemons (which will be already upgraded to new version) ,
 until it isn't upgraded. Once it gets upgraded it will stop crashing.
 I can add a comment before the condition to summarize  this.

Reply:
The "old format" has never been released, so there can't be anyone upgrading from it (at least not upstream).

Reply:
yeah, not in upstream but in downstream it was introduced already.

Reply:
Then it should be patched downstream.  Upstream, this change doesn't make sense -- I would suggesting closing this PR.

Reply:
My main motive was to make sure nothing's breaking and tests are passing so opened a PR here. Closing this as fix is delivered downstream now!